### PR TITLE
Make app work

### PR DESCRIPTION
**Pull request by Gredin67**
## Problem
- Make the app work

## Solution
- The app works, some little bugs removed

## PR Status
- [x] Code finished.
- [ ] Tested with Package_check.
- [ ] Fix or enhancement tested.
- [ ] Upgrade from last version tested.
- [x] Can be reviewed and tested.

## Package_check results
---
*If you have access to [App Continuous Integration for packagers](https://yunohost.org/#/packaging_apps_ci) you can provide a link to the package_check results like below, replacing '-NUM-' in this link by the PR number and USERNAME by your username on the ci-apps-dev. Or you provide a screenshot or a pastebin of the results*

[Build Status/badge/icon)](https://ci-apps-dev.yunohost.org/jenkins/job/APP_ynh%20PR-NUM-%20(USERNAME)/)

### CODE REVIEW

File: manifest.json [hunk 1]
```diff
@@ -27,51 +27,64 @@
...
-					"en": "Choose the user name of the whatsapp bot for which to create a YunoHost and a local synapse server account.",
-					"fr": "Choisissez le nom d'utilisateur du robot whatsapp pour lequel un compte sera créé sur YunoHost et le serveur local synapse."
+					"en": "Choose the local Matrix/Synapse user name of the WhatsApp bot.",
+					"fr": "Choisissez le nom d'utilisateur local Matrix/Synapse du robot WhatsApp."
 				},
 				"example": "whatsappbot",
+				"help": {
+					"en": "Invite @whatsappbot:domain.org from an authorize Matrix account to setup the bridge.",
+					"fr": "Inviter @whatsappbot:domain.org depuis votre compte Matrix principal pour démarrer la passerelle."
+				}
```
Comment (tituspijean) [inline manifest.json]: You are forgetting to add commas when you add new fields.

Comment (Gredin67) [reply manifest.json]: oops, too much copy-pasting... Should I close the pull request, add a commit on my fork, and then resubmit a PR?
Or how should I proceed?

**Comment by Gredin67** [comment]
Well I think now it is ready to be added to the app list. I also tested the remove script. Only update, backup and restore are missing.

File: manifest.json (hunk 1 above)
Comment (Gredin67) [reply manifest.json]: I discovered that pushing to the master branch of my fork adds the commit to the PR. C'est quand meme bien fait!

**Comment by Gredin67** [comment]
@tituspijean what about this PR? Is is running on my test server for some days. I also already PR to add mautrix_whatsapp in the apps list. Still waiting though.
Once the PR will be accepted I plan to create a testing branch to follow the standards in maintaining app.
I still need to understand what should be done for CI tests.

**Comment by tituspijean** [comment]
I still have masquerading issues with your latest code. :(

**Comment by Gredin67** [comment]
You mean this loop in the end? Do you have any hint where to look for?

**Comment by tituspijean** [comment]
Indeed, I get this long loop that ends after some minutes, but I am talking about the bot service completely failing:

```
Apr 24 17:03:48 vps python[24228]: 2020-04-24 17:03:48,942 - synapse.http.matrixfederationclient - 491 - WARNING - federation_transaction_transmission_loop-1368- {PUT-O-1001} [nummer.wang] Request failed: PUT matrix://nummer.wang/_matrix/federation/v1/send/1587731137473: TimeoutError('',)
Apr 24 17:03:58 vps mautrix-whatsapp[25580]: [Apr 24, 2020 17:03:58] [ERROR] Failed to connect to homeserver: msg=Failed to GET JSON to /_matrix/client/r0/account/whoami code=403 wrapped=M_FORBIDDEN: Application service cannot masquerade as this user.. Retrying in 10 seconds...
```

**Comment by Gredin67** [comment]
> Apr 24 17:03:58 vps mautrix-whatsapp[25580]: [Apr 24, 2020 17:03:58] [ERROR] Failed to connect to homeserver: msg=Failed to GET JSON to /_matrix/client/r0/account/whoami code=403 wrapped=M_FORBIDDEN: Application service cannot masquerade as this user.. Retrying in 10 seconds...

Finally I was able to replicate this error. And resolved by defining synapse domain and server_name correctly in the app service registration.
I am currently unable to connect to riot but this is not related as it also happens before I run the install of mautrix_whatsapp

**Comment by Thatoo** [comment]
At install, I have this error : 

ERROR - Impossible d'installer mautrix_whatsapp: Une erreur est survenue dans le script d'installation de l'application
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/moulinette/interfaces/api.py", line 301, in wrapper
    curr_v = params[k]
KeyError: 'label'

**Comment by Thatoo** [comment]
I'm trying to install it. What is the synapse instance number?

**Comment by Thatoo** [comment]
second question. When you ask who could use the robot, the admin only, a uesr, all matrix or a specific server (matrix.org)?

If we want to limit to a specific server, we should write the synapse domain? or the server_name is ok?

**Comment by Gredin67** [comment]
the synapse instance number is 1 if you have only one instance. If you have synapse__2 and you want to setup the bridge on it, then give "2". (not yet correctly implemented)
If you want only one server give the server name I think. If you want to give access to your local synapse instance then say yes to the last question. This whole access rights has to be further tested and improved.

**Review by Thatoo — APPROVED** [review]

**Comment by Gredin67** [comment]
@tituspijean @Thatoo i'm done for today. Somehow the install script does not work with the web install. It seems like the way golang is handled does not fit the web install ideas:
```
export PATH=$PATH:/usr/lib/go-$GO_VERSION/bin
go build
``` 
We get:
``go: cannot use modules with build cache disabled``

File: manifest.json [hunk 2]
```diff
@@ -23,55 +23,70 @@
...
 				"ask": {
-					"en": "Choose the user name of the whatsapp bot for which to create a YunoHost and a local synapse server account.",
-					"fr": "Choisissez le nom d'utilisateur du robot whatsapp pour lequel un compte sera créé sur YunoHost et le serveur local synapse."
+					"en": "Choose a local Matrix/Synapse user name for the WhatsApp bot",
+					"fr": "Choisissez un nom d'utilisateur local Matrix/Synapse pour le robot WhatsApp"
 				},
 				"example": "whatsappbot",
+				"help": {
+					"en": "A system user will be created. Invite @whatsappbot:localsynapse.servername from an authorized Matrix account to start bridging. Give the synapse server name, not the domain.",
+					"fr": "Un utilisateur système sera créé. Inviter @whatsappbot:localsynapse.servername depuis un compte Matrix autorisé pour démarrer une passerelle. Donner le server name, pas le domain"
```
Comment (Thatoo) [inline manifest.json]: Il manque un e à la fin de domaine...
L'erreur est répétée plus bas aussi.

File: manifest.json [hunk 3]
```diff
@@ -23,55 +23,70 @@
...
-				"type": "boolean",
-				"ask": {
-					"en": "Is it a public bot?",
-					"fr": "Est-ce un robot public ?"
+					"en": "Choose Matrix user(s) authorized to bridge with the WhatsApp bot",
+					"fr": "Choisissez le/les compte(s) Matrix autorisés à utiliser le robot WhatsApp"
 				},
-				"default": true,
+				"example": "admin or local or @johndoe:server.name or server.name or *",
+				"default": "admin",
```
Comment (Thatoo) [inline manifest.json:86]: est-ce qu'on ne pourrait pas, pour éviter les erreurs d'écriture proposer des cases à cocher pour
- admin seulement (qui serait coché par défaut et qu'il ne serait pas possible de décocher d'ailleurs)
- serveur local (qui serait le server_name, coché par défaut mais décochable)
- utilisateurs avec un champ pour entrer une liste d'utilisateurs spécifiques (vide par défaut)
- autres serveurs avec un champ pour entrer des serveurs (matrix.org) (vide par défaut)

Si ça ne demande pas trop de dév, c'est ce qui me semblerait le plus clair

File: conf/systemd.service [hunk 4]
```diff
@@ -6,7 +6,7 @@ After=matrix-synapse.service
 Type=simple
 User=__APP__
 WorkingDirectory=/opt/yunohost/__APP__
-ExecStart=/opt/yunohost/__APP__/mautrix-whatsapp -c=/opt/yunohost/__APP__/config.yaml >> /var/log/__APP__/__APP__.log 2>&1
+ExecStart=/opt/yunohost/__APP__/mautrix-whatsapp -c=/opt/yunohost/__APP__/config.yaml >> /var/log/__APP__/log.log 2>&1
```
Comment (Thatoo) [inline conf/systemd.service:9]: mautrix.log ne serait pas plus judicieux?

**Review by Thatoo — CHANGES_REQUESTED** [review]
I suggested few changes. I won't review check_process.default as I'm not sure...

**Review by Thatoo — COMMENTED** [review]
I guess it's ok

File: conf/systemd.service (hunk 4 above)
Comment (Gredin67) [reply conf/systemd.service:9]: C'était la proposition de @tituspijean qui a plus d'expérience que moi dans le packaging. Donc je laisse pour l'instant.

File: manifest.json (hunk 2 above)
Comment (Gredin67) [reply manifest.json]: j'avais laissé en Anglais car ce sont des notions propres à synapse. Je vais voir si he traduis ou si je mets entre guillemets

File: manifest.json (hunk 3 above)
Comment (Gredin67) [reply manifest.json:86]: Je vais laisser comme ca pour l'instant car de toutes facon le script d'installation ne traite pas de liste d'utilisateur. J'ai créé deux issues concernant ces améliorations #4 #5 Surement très simple à implémenter pour quelqu'un qui s'y connait en bash et en sed.

**Pull request merged**